framework/src/engine/bft/method.ts

```diff
@@ -98,27 +98,40 @@ export class BFTMethod {
 		};
 	}
 
-	// This function expects to be called after blockBFTInfos has been inserted for the executiong block.
-	// blockBFTInfos must be inserted first because for the block genesis block + 1, bftVotes.blockBFTInfos will be empty
-	public async currentHeaderImpliesMaximalPrevotes(stateStore: StateStore): Promise<boolean> {
+	public async impliesMaximalPrevotes(
+		stateStore: StateStore,
+		header: { height: number; generatorAddress: Buffer; maxHeightGenerated: number },
+	): Promise<boolean> {
 		const votesStore = stateStore.getStore(MODULE_STORE_PREFIX_BFT, STORE_PREFIX_BFT_VOTES);
 		const bftVotes = await votesStore.getWithSchema<BFTVotes>(EMPTY_KEY, bftVotesSchema);
-		const [currentHeader] = bftVotes.blockBFTInfos;
-		const previousHeight = currentHeader.maxHeightGenerated;
+		// if blockBFTInfos is empty (ie: right after executing genesis block), next block never implies prevotes
+		if (bftVotes.blockBFTInfos.length === 0) {
```

Comment:
This case is differently in the specifications https://github.com/LiskHQ/lips/pull/177/files. Note that the first block after the genesis block can imply prevotes, namely for the block itself.